pkg/config/nodetreemodel/config.go

```diff
@@ -219,6 +242,18 @@ func (c *ntmConfig) GetKnownKeysLowercased() map[string]interface{} {
 	return ret
 }
 
+// BuildSchema is called when Setup is complete, and the config is ready to be used
+func (c *ntmConfig) BuildSchema() {
+	c.ready = true
```

Comment:
Should we lock here or use an atomic bool ?

Reply:
Locking sounds good